### ISSUE TITLE
Can't Recieve on a/p Frequencies

### ISSUE THREAD

**Issue by rnkambara** [bastibl/gr-ieee802-11#209]
**Issue**:
I am using the latest maint-8.0 branch with two *hackrf* devices and the [soapy](https://gitlab.com/librespacefoundation/gr-soapy) SDR and the source and sink.

I am able to transmit and recieve (using wifi_tx and wifi_rx) on all of the */g* frequencies but none of the */a/p* frequencies.

Would you have any idea who some frequencies would be working and some would not?

**Receiver/Transmitter testing**:
I have tried using the [gqrx](http://gqrx.dk/) tool to listen to the frequencies the */a/p* frequencies while running the tx, but it doesn't appear as if anything is picked up.

Similarly using a  *QT GUI Sink* at the beginning of the receiver end, I don't see a change in any of the graphs (i.e. Frequency, waterfall, etc...).

I am however able to see changes occurring and receive messages sent through any of the */g* frequencies.

**Note**:
For this to work even with the */g* frequencies I need to adjust the gain for my TX to at least 24.

**Computer Specs**:
OS: Ubuntu 18.04.1
Memory: 31.2 GB
Processes: Intel® Core™ i7-4770 CPU @ 3.40GHz × 8 

**RX**:
![image](https://user-images.githubusercontent.com/25187959/72746227-8f57c200-3b7f-11ea-9dd5-ee2f20e7d350.png)

**TX**:
![image](https://user-images.githubusercontent.com/25187959/72746261-a5658280-3b7f-11ea-8b20-3734fb042f01.png)

**Comment by bastibl** [comment]
Hi, thanks for the great description of the problem. One thing that'd come to my mind is that it might be related to the gain setting. The same setting might lead to different actual power levels on different frequencies. The HackRF has three gain stages for RX and two for TX. You can try to extend the flowgraph with slides to for each stage and test different values. Here's some more information on the ranges for each stage:

https://github.com/mossmann/hackrf/wiki/FAQ#making-sense-of-gain-settings

**Comment by rnkambara** [comment]
Thank you so much for your reply and recommendation. I set up the gain settings according that that documentation and played around with increasing/decreasing to try in get signals picked up.

Increasing the gain settings on the RX side (all the gain settings) basically just picked up noise. Even when I had my gain settings increased on the TX side it didn't pick up the messages.

Since the messages are being picked up on all of the g frequencies, I think it may be a hardware issue. Possibly with the antennas I am using don't support higher frequencies. Regardless, Feel free to close this issue. If I make any progress getting this to work I will give details at that time. Thank you so much for taking a look!

**Comment by bastibl** [comment]
That's strange. I had some success using it with the HackRF on 5GHz. My guess would be that it's the DC offset, which might be different on different frequencies. I'm not sure what exactly your CorrectIQ block does but maybe it's not enough. I didn't play around much with the parameters but used a pretty steep (and computationally demanding) highpass filter.
Maybe you could give this a shot: https://github.com/bastibl/instant-gnuradio/blob/wlpc/assets/wifi_rx.grc
You can check the signal after the filter if the DC offset is removed and the the signal after the divide block to see if the plateau of the preamble is clearly visible.

**Comment by rnkambara** [comment]
That did it :)

Thank you so much for your help,  I ended up replacing the CorrectIQ block with your FFT Filter on top of tuning the gain values and was eventually able to get transmitting and receiving for each frequency. 

Thanks for mentioning the plateau of the preambles, I didn't think to look for that. Once I noticed that the plateaus were present  it was only a matter of adjusting the gains to get the messages through.

For reference, bellow are example adjustments for a/p/g frequencies I ended up making on the tx/rx for successful messaging.

**1 | 2412.0 | 11g**
```
TX:
IF/VGA: 25
Interval: 50
RF/AMP: 0
```

```
RX:
BB/VGA: 30
IF/LNA: 24
RF/AMP: 0
```

**100 | 5500.0 | 11a**
```
TX:
IF/VGA: 40
Interval: 20
RF/AMP: 14
```

```
RX:
BB/VGA: 30
IF/LNA: 24
RF/AMP: 14
```

**172 | 5860.0 | 11p**
```
TX:
IF/VGA: 45
Interval: 20
RF/AMP: 14
```

```
RX:
BB/VGA: 30
IF/LNA: 24
RF/AMP: 14
```

For more reference, this was working with:

version: gnuadio 3.8
tx/rx devices: hackrf
driver: [soapy](https://gitlab.com/librespacefoundation/gr-soapy) version 3.8
(I used the soapy driver instead of osmosdr drivers since osmosdr is unsupported in gnuradio 3.8)
